In [3]:
import numpy as np
import pandas as pd

# Load Data

In [4]:
hockey = pd.read_csv("data/clean/train.csv"
                     , sep = ',',encoding = "ISO-8859-1",engine='python')
hockey['Age'] = 117 - pd.to_numeric(hockey['Born'].str[0:2])

## Combine the train and test data together for EDA

In [5]:
train = pd.read_csv("data/clean/train.csv"
                     , sep = ',',encoding = "ISO-8859-1",engine='python')

test = pd.read_csv("data/clean/test.csv"
                     , sep = ',',encoding = "ISO-8859-1",engine='python')

test_y = pd.read_csv('data/clean/test_salaries.csv',
                     sep = ',',encoding = "ISO-8859-1",engine='python')
                     
def combine_train_and_test(train_df, test_df, test_response):
    test_df = pd.concat([test_df, test_response], axis = 1)
    return pd.concat([train_df, test_df],ignore_index = True, sort = False)

hockey = combine_train_and_test(train, test, test_y)

## How large is our dataset?

In [8]:
hockey.shape

(874, 154)

<span style='font-size:18px'> 
    Our data contains 874 observations with 154 possible features to be included into a model.<br>
    In order to use the features, we should have an understanding of which features contain missing<br>
    values, and which features do not contain any missing values.
</span>

### Columns with more than one missing value

In [28]:
missing = np.sum(hockey.isna())
missing_df = pd.DataFrame()
missing_df['Column'] = hockey.columns[missing > 0]
missing_df['Num Missing'] = missing[missing > 0].values
missing_df.sort_values('Num Missing', ascending=False).head(15)

Column  Num Missing
0     Pr/St          225
2     DftRd          125
3      Ovrl          125
1     DftYr          125
21  sDist.1           25
13      iCF           11
24      iHA           11
18      iRB           11
15      iSF           11
14      iFF           11
25     iHDf           10
20      iDS           10
62     GS/G            2
9       PDO            2
7       SH%            2

# Distribution of Target Column

<span style='font-size:18px'> 
    Since we are trying to predict the salary of an NHL player, it might be useful for us to visualize<br>
    the distribution of the salary of players.
</span>

First verify that we have no missing values in our target column:

In [33]:
sum(hockey.Salary.isna())

0

In [37]:
unique, counts = np.unique(hockey.Nat, return_counts=True)
dict(zip(unique, counts))

{'AUT': 3,
 'CAN': 409,
 'CHE': 13,
 'CZE': 30,
 'DEU': 5,
 'DNK': 9,
 'FIN': 32,
 'FRA': 4,
 'GBR': 1,
 'HRV': 1,
 'LVA': 1,
 'NOR': 2,
 'RUS': 36,
 'SVK': 11,
 'SVN': 1,
 'SWE': 77,
 'USA': 239}

## Is there a salary difference for players who are undrafted vs. those who were drafted?